src/lsst/cmservice/web_app/static/js/script-grid-utils.js

```diff
@@ -0,0 +1,192 @@
+    else {
+        let resetObj = {
+            fullname: fullname,
+            status: newStatus,
+        };
+
+        resetJson = JSON.stringify(resetObj);
+        url = reset_script_endpoint;
+    }
+
+    // send request
+    let xhr = new XMLHttpRequest();
+    xhr.open('POST', url, true);
+    xhr.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+    xhr.onload = function () {
+        if (xhr.status === 201) {
+            // update script status in grid
+            scriptGridApi
+                .getRowNode(rowIndex)
+                .setDataValue("status", document.querySelector("#targetStatus").selectedOptions[0].text);
+            // refresh grid cells on client side to disable/enable reset button according to new status
+            scriptGridApi.refreshCells({
+                force: true,
+                suppressFlash: false,
+              });
+            resetScriptModal.close();
+        } else {
+            resetScriptModal.close();
+            errorModal.showModal();
+            errorMessage.innerText = JSON.parse(xhr.responseText).detail;
+        }
+    };
+    xhr.onerror = function () {
+        console.error("Error:", xhr.status, xhr.responseText);
+        resetScriptModal.close();
+        errorModal.showModal();
+        errorMessage.innerText = xhr.responseText;
+    };
+    xhr.send(resetJson);
+
+};
+
+const readScriptLog = (log_url) => {
+
+    const xhr = new XMLHttpRequest();
+
+    const url = "/web_app/read-script-log";
+
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            if (xhr.status === 200) {
+                const response = JSON.parse(xhr.responseText);
+                // show file content from response in the textarea
+                document.querySelector('#scriptLogContent').value = response.content;
+                scriptLogModal.showModal();
+            } else {
+                console.error("Error:", xhr.status, xhr.responseText);
+                errorModal.showModal();
+                errorMessage.innerText = JSON.parse(xhr.responseText).detail;
+            }
+        }
+    };
+
+    xhr.open("POST", url, true);
+
+    xhr.setRequestHeader("Content-Type", "application/json");
+
+    const requestData = JSON.stringify({ log_path: log_url});//'/home/eman/PycharmProjects/cm-service/output/archive/HSC_DRP-Prod/test_htcondor/step1/group1/job_000/bps_submit_000_bps_config.yaml' });
+    xhr.send(requestData);
+}
+
+class ResetButtonCellRenderer {
+    init(params) {
+        this.params = params;
+        this.eGui = document.createElement('button');
+        // change button text and also enable/disable according to status
+        let currentStatus = params.data.status;
+        if(currentStatus === "FAILED" || currentStatus === "REJECTED") {
+            this.eGui.innerText = 'Reset';
+        } else if(currentStatus === "REVIEWABLE") {
+            this.eGui.innerText = 'Review';
+        } else {
```

Comment:
Can any of this UI logic be given over to HTMX to control? Can it be done on the server as part of rendering?

Reply:
Nope. This is the part where HTMX didn't work because of the auto refresh. As we discussed, it word work if the htmx tags are outside the auto-refreshed section. Unfortinately this cannot be achieved with the buttons inside the grid